Sources/TuistDependencies/Carthage/CarthageInteractor.swift

```diff
@@ -0,0 +1,204 @@
+
+public final class CarthageInteractor: CarthageInteracting {
+    private let fileHandler: FileHandling
+    private let carthageCommandGenerator: CarthageCommandGenerating
+    private let cartfileContentGenerator: CartfileContentGenerating
+    private let carthageFrameworksInteractor: CarthageFrameworksInteracting
+
+    public init(
+        fileHandler: FileHandling = FileHandler.shared,
+        carthageCommandGenerator: CarthageCommandGenerating = CarthageCommandGenerator(),
+        cartfileContentGenerator: CartfileContentGenerating = CartfileContentGenerator(),
+        carthageFrameworksInteractor: CarthageFrameworksInteracting = CarthageFrameworksInteractor()
+    ) {
+        self.fileHandler = fileHandler
+        self.carthageCommandGenerator = carthageCommandGenerator
+        self.cartfileContentGenerator = cartfileContentGenerator
+        self.carthageFrameworksInteractor = carthageFrameworksInteractor
+    }
+
+    public func install(dependenciesDirectory: AbsolutePath, method: InstallDependenciesMethod, dependencies: [CarthageDependency]) throws {
+        logger.info("Start installing Carthage dependencies.", metadata: .subsection)
+
+        // check availability of `carthage`
+        guard canUseSystemCarthage() else {
+            throw CarthageInteractorError.carthageNotFound
+        }
+
+        // determine platforms
+        let platforms: Set<Platform> = dependencies
+            .reduce(Set<Platform>()) { platforms, dependency in platforms.union(dependency.platforms) }
+
+        try fileHandler.inTemporaryDirectory { temporaryDirectoryPath in
+            // prepare paths
+            let pathsProvider = PathsProvider(dependenciesDirectory: dependenciesDirectory, temporaryDirectoryPath: temporaryDirectoryPath)
+
+            // prepare for installation
+            try prepareForInstallation(pathsProvider: pathsProvider, dependencies: dependencies)
+
+            // create `carthage` shell command
+            let command = carthageCommandGenerator.command(method: method, path: temporaryDirectoryPath, platforms: platforms)
+
+            // run `carthage`
+            try System.shared.runAndPrint(command)
+
+            // post intallation actions
+            try postInstallationActions(pathsProvider: pathsProvider)
+        }
+
+        logger.info("Successfully installed Carthage dependencies.", metadata: .success)
+    }
+
+    // MARK: - Installation
+
+    private func prepareForInstallation(pathsProvider: PathsProvider, dependencies: [CarthageDependency]) throws {
+        // copy build directory from previous run if exist
+        if fileHandler.exists(pathsProvider.destinationCarthageBuildDirectory) {
+            try copyDirectory(from: pathsProvider.destinationCarthageBuildDirectory, to: pathsProvider.temporaryCarthageBuildDirectory)
+        }
+
+        // copy `Cartfile.resolved` from previous run if exist
+        if fileHandler.exists(pathsProvider.destinationCarfileResolvedPath) {
```

Comment:
If I'm not wrong, I don't think Carthage uses the `Cartfile.resolved` file to determine what frameworks need to be fetched. If that's the case _(which I'd recommend validating)_, I don't think we need to copy this file.

Reply:
You have right. Im removing this step.